### ISSUE TITLE
Adding Visual Studio Developer Tools - Incorrect paths

### ISSUE THREAD

**Issue by fdacumos** [microsoft/terminal#12533]
### Windows Terminal version

1.12.10334.0

### Windows build number

10.0.19042.0

### Other Software

Visual Studio 2019
Visaul Studio 2022

### Steps to reproduce

1. Install Visual Studio 2019
2. Install Windows Terminal
3. During the install, choose to add the Visual Studio Developer Tools as a Profile (Developer PowerShell for VS 2019)
4. Install Visual Studio 2022
5. Attempt to open up the "Developer PowerShell for VS 2019" WT Profile

### Expected Behavior

When going to Windows Terminal, I should be able to open up my VS 2019 Developer Prompt by selecting "Developer PowerShell for VS 2019".  However, it errors out and the icon is also missing.

### Actual Behavior

I get the error:
```
Import-Module: Assembly with same name is already loaded
Enter-VsDevShell: Cannot convert 'System.Object[]' to the type 'System.String' required by parameter 'VsInstallPath'. Specified method is not supported.
```

Reasons:
The Problem is with the Windows Terminal Profile settings.  For that WT Profile, under commandLine it shows:
```
"commandline": "C:\\Program Files\\PowerShell\\7\\pwsh.exe -noe -c \"&{$vsPath = &(Join-Path ${env:ProgramFiles(x86)} '\\Microsoft Visual Studio\\Installer\\vswhere.exe') -property installationpath; Import-Module (Join-Path $vsPath 'Common7\\Tools\\Microsoft.VisualStudio.DevShell.dll'); Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation}\"; cd ~",
```
If you have multiple versions of Visual Studio installed, the `vswhere.exe` will return an Array of items, so the PowerShell to register is incorrect.  It needs to reference the first element of the `$vsPath` variable.  This can be manually fixed by editing the settings and changing the line to:
```
"commandline": "C:\\Program Files\\PowerShell\\7\\pwsh.exe -noe -c \"&{$vsPath = &(Join-Path ${env:ProgramFiles(x86)} '\\Microsoft Visual Studio\\Installer\\vswhere.exe') -property installationpath; Import-Module (Join-Path $vsPath[0] 'Common7\\Tools\\Microsoft.VisualStudio.DevShell.dll'); Enter-VsDevShell -VsInstallPath $vsPath[0] -SkipAutomaticLocation}\"; cd ~",
```

**Missing Icon**
Also, for the missing Icon, the default install references the Icon under the Visual Studio "Enterprise".
```
"%ProgramFiles(x86)%\\Microsoft Visual Studio\\2019\\Enterprise\\Common7\\IDE\\Assets\\VisualStudio.70x70.contrast-standard_scale-180.png",
```

If you have Visual Studio "Professional" installed, the icon will break.  Simply fix this by changing the line to:
```
"%ProgramFiles(x86)%\\Microsoft Visual Studio\\2019\\Professional\\Common7\\IDE\\Assets\\VisualStudio.70x70.contrast-standard_scale-180.png",
```

**Comment by heaths** [comment]
That's not the generator. The generated uses the COM APIs (the same ones vswhere uses) and emits the correct command line without using vswhere, which would be redundant. Here's a generated command line on my machine for VS2019:

For cmd:
```
cmd.exe /k "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\Tools\VsDevCmd.bat" -arch=x64 -host_arch=x64
```

For PowerShell:
```
powershell.exe -NoExit -Command "&{Import-Module """C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\Tools\Microsoft.VisualStudio.DevShell.dll"""; Enter-VsDevShell c34e3a7d -SkipAutomaticLocation -DevCmdArguments """-arch=x64 -host_arch=x64"""}"
```

That the icon is the generic and should be VS was also suggested on Twitter I saw but, honestly, the icons in Windows Start are the same as in terminal.

<img width="239" alt="image" src="https://user-images.githubusercontent.com/1532486/154779813-0b79d563-7223-4052-868e-cd9921f81830.png">

We don't have any assets in VS currently for VS-branded icons that I'm aware of. The icons have been like this as long as I can remember. If people feel strongly they should be different than what VS installs, I recommend a separate bug that will have to go through VS triage to get produced official, I imagine.

**Comment by fdacumos** [comment]
@heaths - for the "missing icon" it wasn't the actual icon that was the issue. The problem I noticed was that the path that was referenced pointed to `Enterprise`.  However, I have VS `Professional` installed and it didn't pick up the correct path.  Once I made the change in the settings from `Enterprise` to `Professional` the icon picked up correctly.

@DHowett - I'll have to run the install wizard again when I'm on that machine to show what I selected.  But it's the option that Kayla mentioned [here](https://devblogs.microsoft.com/commandline/windows-terminal-preview-1-12-release/) about "Profiles for Visual Studio Developer Command Prompt (VS2017+) and Visual Studio Developer PowerShell (VS2019.2+) will now be automatically generated"

**Comment by heaths** [comment]
I don't recall the exact timeframe, but the original generators for the VS profiles weren't ideal and I completely changed them. It sounds like you have the old profiles. I'd open up your settings.json file, delete them, and let them generate again. The paths I showed above is what is correct and shipped with 1.12 GA.

**Comment by zadjii-msft** [comment]
@fdacumos did deleting the profiles and having them get regenerated for you work/? (You might also have to clear them out of `state.json`, but I can help with that if need be)
